File: CppPractice/CppPractice/di_string_match.cpp
#pragma once
#include "di_string_match.h"

using namespace std;

vector<int> Solution::diStringMatch(string S) {
	int length = S.length();
	int min = 0;
	int max = length;
	vector<int> result;
	for (int i = 0; i < length; i++) {
		if (S[i] == 'I') {
			result.push_back(min++);
		}
		else if (S[i] == 'D') {
			result.push_back(max--);
		}
	}
	result.push_back(min++);
	return result;


}

File: JavaPractice/src/leetcode/DiStringMatch.java
package leetcode;

public class DiStringMatch {
    public int[] diStringMatch(String S) {
        int[] N = new int[S.length()+1];
        int max = S.length();
        int min = 0;
        for(int i =0; i < S.length(); i++){
            char X = S.charAt(i);
            if (X == 'I') {
                N[i] = min;
                min++;
            } else if(X == 'D'){
                N[i] = max;
                max--;
            }
        }
        N[S.length()] = min++;
        return N;
    }
}
